### PR TITLE
Fix: The img field in the mint transaction is mandatory.

### PR DESCRIPTION
**Pull request by btcopenstamp**
Currently, the SRC101 deploy transaction requires specifying imglp and imgf. The img field value can be directly derived from the deploy transaction information, making it unnecessary to set the img field in the mint transaction (even if it is set, it will be ignored). This approach significantly reduces the size of mint transactions.

### CODE REVIEW

File: indexer/src/index_core/src101.py [hunk 1]
```diff
@@ -543,15 +545,12 @@ def handle_mint(self):
                     recipient_nvalue=self.src101_dict.get("destination_nvalue"),
                 )
                 return
-            # check img
+            # set img
             if self.imglp and self.imgf:
+                if len(self.src101_dict.get("tokenid")) > len(self.src101_dict.get("img")):
```
Comment (vikkkko) [inline indexer/src/index_core/src101.py:550]: Why don't you just change it to !=

Comment (btcopenstamp) [reply indexer/src/index_core/src101.py:550]: When the number of `img` exceeds the number of `tokenid`, we still consider the transaction valid. To enhance understanding, we will adopt a more user-friendly approach to accommodate more scenarios.

**Review by superboyiii — APPROVED** [review]

**Comment by btcopenstamp** [comment]
Hi @reinamora137, could you please review and merge the PR at your earliest convenience? We need it in production before launching our domain minting tool. Thanks!

**Comment by superboyiii** [comment]
> Since the proposal has taken effect, if we want to modify it, we need to set a height to enable this optimization. And retain the original logic.

Yes, we need a fork height, otherswise data may conflict when resync indexer.

**Comment by reinamora137** [comment]
> > Since the proposal has taken effect, if we want to modify it, we need to set a height to enable this optimization. And retain the original logic.
> 
> Yes, we need a fork height, otherswise data may conflict when resync indexer.

If you want to set a block activation we can make this live

**Comment by btcopenstamp** [comment]
> Since the proposal has taken effect, if we want to modify it, we need to set a height to enable this optimization. And retain the original logic.

@vikkkko @superboyiii It's compatible. Try running this branch to index the bitname domain from the SRC101 activation block and compare the results with the current release.

**Comment by btcopenstamp** [comment]
The following two mint transactions have img fields that do not meet the requirements and are currently considered invalid. We will update this PR and set a predefined upgrade block height (BTC_SRC101_IMG_OPTIONAL_BLOCK=872500, adjustable as needed).

ffd10ada1715fd53d427442613ceb9e910ddceea61e26fc4ab87e870468a3c61
ee7d5e1839a2e5df93ba13fd6211f19f24d5d0d5f390617e5645b0d47cb1be4d

**Comment by btcopenstamp** [comment]
PR updated with an activation block(BTC_SRC101_IMG_OPTIONAL_BLOCK=872200, adjustable as needed). 

Our mint functionality relies on this PR, so we hope it can be merged and deployed as soon as possible.

@reinamora137 @vikkkko @superboyiii

**Comment by reinamora137** [comment]
closing. rebased in a local branch: https://github.com/stampchain-io/btc_stamps/pull/470

**Pull request closed without merging**